conans/test/functional/remote/test_request_headers.py

```diff
@@ -0,0 +1,160 @@
+
+
+@parameterized_class([{"revs_enabled": True}, {"revs_enabled": False}, ])
+class RequestHeadersTestCase(unittest.TestCase):
+    """ Conan adds a header with the settings used to compute the package ID """
+
+    profile = textwrap.dedent("""
+        [settings]
+        os=Macos
+        arch=x86_64
+        compiler=apple-clang
+        compiler.version=11.0
+        compiler.libcxx=libc++
+        build_type=Release
+    """)
+
+    conanfile = GenConanfile().with_settings('os', 'arch', 'compiler') \
+        .with_option('opt1', [True, False]) \
+        .with_option('shared', [True, False]) \
+        .with_default_option('opt1', True) \
+        .with_default_option('shared', False)
+
+    def setUp(self):
+        test_server = TestServer(users={"user": "mypass"})
+        self.servers = {"default": test_server}
+        t = TestClient(servers=self.servers, users={"default": [("user", "mypass")]})
+        t.save({'conanfile.py': self.conanfile,
+                'profile': self.profile})
+        t.run('create conanfile.py name/version@user/channel --profile:host=profile')
+        t.run('upload name/version@user/channel --all')
+
+    def _get_header(self, requester, header_name):
+        hits = sum([header_name in headers for _, headers in requester.requests])
+        self.assertEquals(hits, 2 if self.revs_enabled else 1)
+        for url, headers in requester.requests:
+            if header_name in headers:
+                if self.revs_enabled:
+                    self.assertTrue(url.endswith('/latest'), msg=url)
+                else:
+                    self.assertTrue(url.endswith('/download_urls'), msg=url)
+                return headers.get(header_name)
+
+    def _assert_settings_headers(self, settings_header, compiler_version='11.0'):
+        # It takes only the values that are relevant to the recipe
+        self.assertListEqual(
+            sorted(['os', 'arch', 'compiler', 'compiler.version', 'compiler.libcxx']),
+            sorted([it.split('=', 1)[0] for it in settings_header.split(';')]))
+        self.assertIn('os=Macos', settings_header)
+        self.assertIn('arch=x86_64', settings_header)
+        self.assertIn('compiler=apple-clang', settings_header)
+        self.assertIn('compiler.libcxx=libc++', settings_header)
+        self.assertIn('compiler.version={}'.format(compiler_version), settings_header)
+        self.assertNotIn('build_type', settings_header)
+
+    def _assert_options_headers(self, options_header, shared_value='False'):
+        self.assertListEqual(['shared'], [it.split('=', 1)[0] for it in options_header.split(';')])
+        self.assertIn('shared={}'.format(shared_value), options_header)
+
+    def _get_test_client(self):
+        t = TestClient(requester_class=RequesterClass, servers=self.servers,
+                       users={"default": [("user", "mypass")]})
+        t.run('config set general.revisions_enabled={}'.format('1' if self.revs_enabled else '0'))
```

Comment:
This is running the same test 2 times, right? Once with TESTING_REVISIONS_ENABLED and once without it, then overriding the testing one. I would suggest using the ``cache.config.revisions_enabled`` to check what is the current value defined by testing.